iot/iot-service-base/src/main/java/io/enmasse/iot/service/base/infinispan/device/DeviceKey.java

```diff
@@ -41,7 +41,7 @@ protected DeviceKey() {}
      * @param tenantId the id of the tenant owning the registration key.
      * @param deviceId the id of the device being registered.
      */
-    private DeviceKey(final String tenantId, final String deviceId) {
+    public DeviceKey(final String tenantId, final String deviceId) {
```

Comment:
The idea is to not use the constructor, but the `deviceKey` method, in order to do some checks before.